modules/samples/watertank/component/ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.watertank.device-view/device-view.js

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+function onRequest(context) {
+    var log = new Log("cdmf.unit.device.type.watertank.device-view.js");
+    var deviceType = context.uriParams.deviceType;
+    var deviceId = request.getParameter("id");
+    var autoCompleteParams = [
+        {"name": "deviceId", "value": deviceId}
+    ];
+
+    if (deviceType != null && deviceType != undefined && deviceId != null && deviceId != undefined) {
+        var deviceModule = require("/app/modules/device.js").deviceModule;
+        var device = deviceModule.viewDevice(deviceType, deviceId);
+        if (device && device.status != "error") {
```

Comment:
How bracket notation could be more reliable? It is all about the language semantics. Using dot notation is permitted in JS as long as attribute name is a valid identifier.
http://stackoverflow.com/a/4968448/1391687